wikidatarefisland/run.py

```diff
@@ -64,6 +64,20 @@ def main(argv, filepath):
         simple_pump.run(pipe, args.input_path, args.output_path)
         return
 
+    if 'item_analysis' == args.step:
+        whitelisted_ext_ids = storage.get(args.side_service_input_path)
+        pipe = pipes.ItemStatisticalAnalysisPipe(
+            whitelisted_ext_ids,
+            config.get('minimum_repetitions_for_item_values'),
+            config.get('maximum_noise_ratio_for_item_values')
+        )
+        observer_pump = pumps.ObserverPump(storage)
+        observer_pump.run(pipe, args.input_path, '-')
+        mapping = pipe.get_mapping()
```

Comment:
This pattern is strange to me... we're running a pump to update the pipe class itself. I'm not fully convinced that this Analysis class really needs all of this, after all this `get_mapping()` method could never really be called without the observation pump, so they don't make sense to me as separate classes.

Reply:
I see but the observer pump is stateless and makes the pipe running, otherwise the pipe itself needs to know about storage which is something I want avoid here.

Reply:
Does the observerpump need to write to disk? currently it's just reading from disk. and as this analyzer class is not really a "Pipe" per se, I don't see why it couldn't have storage as a direct dependency.

Reply:
Adding storage as a dependency makes the pipe to be coupled to the class and makes testing harder. Also even if this is not a "real" pipe, I would like to keep it as similar as possible to other ones for sake of consistency to make the code easier to understand

Reply:
coolio :) Thanks for the explenation